### PR TITLE
chore: this makes the build suck less; introduces netlify deploy on PR

### PR DESCRIPTION
**Pull request by DSchau**
This PR internalizes the build tooling (way clearer what's going on), bumps some versions of things, etc. etc.

In particular:

- [x] Upgrades to latest versions of webpack/webpack-dev-server
- [x] Upgrades to latest versions of react/react-dom
- [x] Bumps to latest version of Reveal.js
- [x] Internalizes Webpack configuration to _this_ repo instead of in some obscure package (my fault...)
- [x] Introduces Netlify with netlify.toml
- [x] Keeps (most???) things working as is; including build, split CSS file, Reveal.js plugins (slide notes), etc.
- [x] Bumps version to Node 8/LTS for workshop set-up

### CODE REVIEW

**Comment by DSchau** [comment]
Deploy preview for *react-redux-object-partners* ready!

Built with commit 3c877c4741e5aba8380e47b0a3fedfb19cf3b26b

https://deploy-preview-19--react-redux-object-partners.netlify.com

File: slidedeck/package.json [hunk 1]
```diff
@@ -9,37 +9,49 @@
...
+    "babel": "~6.23.0",
+    "babel-loader": "~7.1.4",
+    "babel-preset-env": "~1.6.1",
+    "babel-preset-react": "~6.24.1",
+    "cross-env": "~5.1.4",
+    "css-loader": "~0.28.11",
+    "cssnano": "~3.10.0",
     "del-cli": "~0.2.1",
-    "ghpages": "~0.0.10",
-    "object-partners-presentation": "objectpartners/object-partners-presentation-config#1.0.1",
```
Comment (tmartensen) [inline slidedeck/package.json]: Is this not published on `npm` like before?

File: slidedeck/.babelrc.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+  presets: [
+    [
+      'babel-preset-env',
+      {
+        targets: {
+          browsers: ['>0.25%', 'not ie 11', 'not op_mini all']
```
Comment (tmartensen) [inline slidedeck/.babelrc.js]: No IE 11? What's preventing that?

File: slidedeck/src/components/slide-deck/slide-deck.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  render() {
+    const { slides } = this.props;
+    return (
+      <div className="reveal">
+        <div className="slides">
+          <section data-state="title">
+            <h1>React</h1>
+            <h2>HDC - September 6, 2017</h2>
```
Comment (tmartensen) [inline slidedeck/src/components/slide-deck/slide-deck.js]: We should really parameterize this.

File: slidedeck/.babelrc.js (hunk 2 above)
Comment (DSchau) [reply slidedeck/.babelrc.js]: Thought leading ™️ 

https://jamie.build/last-2-versions

File: slidedeck/package.json (hunk 1 above)
Comment (DSchau) [reply slidedeck/package.json]: I published it on NPM the whole time, we were just using the Github tag. I think I prefer using the variant on NPM.

File: slidedeck/src/components/slide-deck/slide-deck.js (hunk 3 above)
Comment (DSchau) [reply slidedeck/src/components/slide-deck/slide-deck.js]: Agreed. Environment variables maybe... ?

**Pull request merged**